party_menu/party_menu.lua

```diff
@@ -0,0 +1,469 @@
+            -- Don't want to change debuffs here
+        else
+            party_display_strings[i].debuffs = ''
+        end
+    end
+end
+update_party_status_strings('both')
+
+local party_window_states = {}
+
+local create_party_window_states = function()
+    for i = 1, 6 do
+        party_window_states[i] = {
+            style = 'chromeless',
+            x = options.global_window_state.x + outer_padding,
+            width = options.global_window_state.width - 2 * outer_padding,
+            height = row_window_height,
+            y = options.global_window_state.y + i * outer_padding + (i - 1) * row_window_height,
+        }
+    end
+end
+create_party_window_states()
+
+local update_ui_dimensions = function()
+    row_window_height = options.text_size + upper_inner_padding + lower_inner_padding
+    global_window_state.y = options.global_window_state.y - (party.alliance.party_1_count - 1) * (row_window_height + outer_padding)
+    global_window_state.x = options.global_window_state.x
+    global_window_state.width = options.global_window_state.width
+    global_window_state.height = party.alliance.party_1_count * (row_window_height + outer_padding) + outer_padding
+    for i, row_window in ipairs(party_window_states) do
+        row_window.x = options.global_window_state.x + outer_padding
+        row_window.width = options.global_window_state.width - 2 * outer_padding
+        row_window.y = global_window_state.y + outer_padding * i + row_window_height * (i - 1)
+        row_window.height = row_window_height
+    end
+end
+
+local check_target_for_row_highlighting = function()
+    local entity = target.t or target.st
+    for i = 1, 6 do
+        if entity and party[i] then
+            if entity.name == party[i].name then
+                party_window_states[i].color = ui.color.rgb(30, 30, 30)
+            else
+                party_window_states[i].color = ui.color.black
+            end
+        else
+            party_window_states[i].color = ui.color.black
+        end
+    end
+end
+
+-- Player zone out packet
+packets.incoming[0x00B]:register(function()
+    zone_out_time = os.clock()
+    zoning = true
+end)
+
+-- 0x0DF appears to change when player hp/mp/tp changes
+packets.incoming[0x0DF]:register(function()
+    update_party_name_hp_mp_tp_strings('party_only')
+end)
+
+-- This seems to change when buffs on self change, also when party members zone
+packets.incoming[0x037]:register(function()
+    update_party_status_strings('player_only')
+end)
+
+-- Party member updates, joining/leaving party etc
+packets.incoming[0x0DD]:register(function()
+    update_party_name_hp_mp_tp_strings('both')
+    update_party_status_strings('both')
+end)
+
+-- Party buffs update packet
+packets.incoming[0x076]:register(function()
+    update_party_status_strings('party_only')
+end)
+
+-- This seems to be the most reliable way (I have found so far) to track party member zoning
```

Comment:
If a party member zones, you'll receive one 0x0C8 chunk followed by one 0x0DD per party member. There's a zone field in the structure of both. If it's zero, they're in the same zone as you.
You could use whichever is convenient. 0x0C8 is pretty much 18 0x0DD chunks mashed together, plus some extra information but minus the strings for party member names.

0x067 is used for other things. You receive a ton of them any time you have a pet out, for instance.

Reply:
Using 0x0C8 definitely fixed some slowdown problems, thanks for the tip!